File: src/types/currency.ts
// File: src/types/currency.ts

export interface ExchangeRate {
  id: number;
  from_currency: string;
  to_currency: string;
  rate: number;
  effective_date: string;
  created_at: string;
  updated_at: string;
}

export interface SetExchangeRateInput {
  from_currency: string;
  to_currency: string;
  rate: number;
  effective_date: string;
}

export interface CurrencyConversion {
  from_currency: string;
  to_currency: string;
  original_amount: number;
  converted_amount: number;
  rate_used: number;
  rate_date: string;
}

export interface SupportedCurrency {
  code: string;
  name: string;
  symbol: string;
}

export interface ExchangeRateSummary {
  from_currency: string;
  to_currency: string;
  latest_rate: number;
  latest_date: string;
  rate_count: number;
}

export interface ConvertedBalance {
  account_id: number;
  account_name: string;
  original_currency: string;
  original_balance: number;
  primary_currency: string;
  converted_balance: number;
  rate_used: number; // 0.0 means no rate found
}

File: src-tauri/src/models/currency.rs
// File: src-tauri/src/models/currency.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ExchangeRate {
    pub id: i64,
    pub from_currency: String,
    pub to_currency: String,
    pub rate: f64,
    pub effective_date: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct SetExchangeRateInput {
    pub from_currency: String,
    pub to_currency: String,
    pub rate: f64,
    pub effective_date: String,
}

#[derive(Debug, Serialize)]
pub struct CurrencyConversion {
    pub from_currency: String,
    pub to_currency: String,
    pub original_amount: f64,
    pub converted_amount: f64,
    pub rate_used: f64,
    pub rate_date: String,
}

#[derive(Debug, Serialize)]
pub struct SupportedCurrency {
    pub code: String,
    pub name: String,
    pub symbol: String,
}

#[derive(Debug, Serialize)]
pub struct ExchangeRateSummary {
    pub from_currency: String,
    pub to_currency: String,
    pub latest_rate: f64,
    pub latest_date: String,
    pub rate_count: i64,
}
